### DISCUSSION TITLE
How do I download a single file from a git lfs repo using python?

### DISCUSSION THREAD

**Discussion by shaked571** [git-lfs/git-lfs#4802] [Q&A]
Hi,
I am tracking some `.csv` and `.json` file using LFS in my repo, and I wish to download a single file.

Because I'm running on a cluster, which I need to initial every time, I wish that the download would happen within the code. 

Im getting back
``` 
version https://git-lfs.github.com/spec/v1
oid sha256:376e3b8d27dfa7ca45c68d28466a2327832a08a92ef252502588937eeb78c8943
size  175233
```
which from what I understood is a pointer to the data.
my lfs env looks like:
``` 
git-lfs/3.0.2 (GitHub; darwin amd64; go 1.17.2)
git version 2.30.1 (Apple Git-130)

Endpoint=https://gecgithub01.walmart.com/usr/datasets.git/info/lfs (auth=none)
  SSH=git@gecgithub01.walmart.com:usr/datasets.git
LocalWorkingDir=/Users/usr/Dev/datasets
LocalGitDir=/Users/s0g0a87/Dev/datasets/.git
LocalGitStorageDir=/Users/usr/Dev/datasets/.git
LocalMediaDir=/Users/usr/Dev/datasets/.git/lfs/objects
LocalReferenceDirs=
TempDir=/Users/s0g0a87/Dev/datasets/.git/lfs/tmp
ConcurrentTransfers=8
TusTransfers=false
BasicTransfersOnly=false
SkipDownloadErrors=false
FetchRecentAlways=false
FetchRecentRefsDays=7
FetchRecentCommitsDays=0
FetchRecentRefsIncludeRemotes=true
PruneOffsetDays=3
PruneVerifyRemoteAlways=false
PruneRemoteName=origin
LfsStorageDir=/Users/usr/Dev/datasets/.git/lfs
AccessDownload=none
AccessUpload=none
DownloadTransfers=basic,lfs-standalone-file,ssh
UploadTransfers=basic,lfs-standalone-file,ssh
GIT_EXEC_PATH=/Library/Developer/CommandLineTools/usr/libexec/git-core
git config filter.lfs.process = "git-lfs filter-process"
git config filter.lfs.smudge = "git-lfs smudge -- %f"
git config filter.lfs.clean = "git-lfs clean -- %f"

```
Where should I go from now?(tried to work also with python's packages such as `giftless_client` but I didnt manage to make the download method to work)

**Comment by rageshkrishna** [comment]
I know it's a _really_ late reply to your question, but I've been trying to figure this out myself today. It turns out that if you have a pointer file, you can run it through `git lfs smudge` to get the contents.

For example, let's say you have a file `foo.pointer` with the following contents (I just copied it from your post above):

```
version https://git-lfs.github.com/spec/v1
oid sha256:376e3b8d27dfa7ca45c68d28466a2327832a08a92ef252502588937eeb78c8943
size  175233
```

You can now do this:

```
cat foo.pointer | git lfs smudge
```

This will download the file if it's hasn't already been done earlier, place it in your local lfs directory (usually somewhere in `.git/lfs/objects`), and also dump the contents of the file to stdout.
